[2024-10-17T21:39:13.981+0100] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-10-17T21:39:13.986+0100] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: taskgroup_and_labels_pipeline.Grouping.groupby_region_smoker scheduled__2024-10-16T00:00:00+00:00 [queued]>
[2024-10-17T21:39:13.988+0100] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: taskgroup_and_labels_pipeline.Grouping.groupby_region_smoker scheduled__2024-10-16T00:00:00+00:00 [queued]>
[2024-10-17T21:39:13.988+0100] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2024-10-17T21:39:13.992+0100] {taskinstance.py:2888} INFO - Executing <Task(PythonOperator): Grouping.groupby_region_smoker> on 2024-10-16 00:00:00+00:00
[2024-10-17T21:39:13.994+0100] {standard_task_runner.py:72} INFO - Started process 4652 to run task
[2024-10-17T21:39:13.997+0100] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'taskgroup_and_labels_pipeline', 'Grouping.groupby_region_smoker', 'scheduled__2024-10-16T00:00:00+00:00', '--job-id', '260', '--raw', '--subdir', 'DAGS_FOLDER/taskgroups_edgelabels.py', '--cfg-path', '/var/folders/nl/1x330z2x7yzb268yb0bbvjcr0000gn/T/tmpqgni_mgz']
[2024-10-17T21:39:13.998+0100] {standard_task_runner.py:105} INFO - Job 260: Subtask Grouping.groupby_region_smoker
[2024-10-17T21:39:14.012+0100] {task_command.py:467} INFO - Running <TaskInstance: taskgroup_and_labels_pipeline.Grouping.groupby_region_smoker scheduled__2024-10-16T00:00:00+00:00 [running]> on host mujtabas-macbook-pro.local
[2024-10-17T21:39:14.031+0100] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='ali' AIRFLOW_CTX_DAG_ID='taskgroup_and_labels_pipeline' AIRFLOW_CTX_TASK_ID='Grouping.groupby_region_smoker' AIRFLOW_CTX_EXECUTION_DATE='2024-10-16T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-10-16T00:00:00+00:00'
[2024-10-17T21:39:14.031+0100] {taskinstance.py:731} INFO - ::endgroup::
[2024-10-17T21:39:14.049+0100] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/Users/mujtabaalimalik/airflow/dags/taskgroups_edgelabels.py", line 66, in groupby_region_smoker
    region_df.to_csv('/Users/loonycorn/airflow/output/grouped_by_region.csv', index=False)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/pandas/core/generic.py", line 3772, in to_csv
    return DataFrameRenderer(formatter).to_csv(
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/pandas/io/formats/format.py", line 1186, in to_csv
    csv_formatter.save()
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/pandas/io/formats/csvs.py", line 240, in save
    with get_handle(
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/pandas/io/common.py", line 737, in get_handle
    check_parent_directory(str(handle))
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/pandas/io/common.py", line 600, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: '/Users/loonycorn/airflow/output'
[2024-10-17T21:39:14.056+0100] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=taskgroup_and_labels_pipeline, task_id=Grouping.groupby_region_smoker, run_id=scheduled__2024-10-16T00:00:00+00:00, execution_date=20241016T000000, start_date=20241017T203913, end_date=20241017T203914
[2024-10-17T21:39:14.059+0100] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-10-17T21:39:14.059+0100] {standard_task_runner.py:124} ERROR - Failed to execute job 260 for task Grouping.groupby_region_smoker (Cannot save file into a non-existent directory: '/Users/loonycorn/airflow/output'; 4652)
Traceback (most recent call last):
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/Users/mujtabaalimalik/airflow/dags/taskgroups_edgelabels.py", line 66, in groupby_region_smoker
    region_df.to_csv('/Users/loonycorn/airflow/output/grouped_by_region.csv', index=False)
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/pandas/core/generic.py", line 3772, in to_csv
    return DataFrameRenderer(formatter).to_csv(
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/pandas/io/formats/format.py", line 1186, in to_csv
    csv_formatter.save()
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/pandas/io/formats/csvs.py", line 240, in save
    with get_handle(
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/pandas/io/common.py", line 737, in get_handle
    check_parent_directory(str(handle))
  File "/opt/miniconda3/envs/airflow/lib/python3.8/site-packages/pandas/io/common.py", line 600, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: '/Users/loonycorn/airflow/output'
[2024-10-17T21:39:14.104+0100] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-10-17T21:39:14.109+0100] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-10-17T21:39:14.110+0100] {local_task_job_runner.py:245} INFO - ::endgroup::
